### ISSUE TITLE
Black screen issue still exists.  Weather overlay not appearing, Aerial (legacy screensaver not requesting location service.

### ISSUE THREAD

**Issue by RGM1956** [JohnCoates/Aerial#1340]
#### General troubleshooting tips

Before logging an issue please check that:
- [x] You have the latest version installed (There may be a beta version that fixes your issue), see here for the latest releases and bug fixes: https://github.com/JohnCoates/Aerial/releases
- [x] Your issue isn't already mentioned in our [issues](https://github.com/JohnCoates/Aerial/issues). You may find a workaround there or a similar request already made.
- [ ] Your problem isn't mentioned in the [troubleshooting page](https://github.com/JohnCoates/Aerial/blob/master/Documentation/Troubleshooting.md).

If none of this fixes your issue, tell us about the problem you are experiencing or the feature you'd like to request.

#### Required information
In order to help us sort your issue, we ask that you provide the following information:
- [x] Mac model: 
- [x] macOS version: 
- [ ] Monitor setup: 

If appropriate, please enable `Debug mode` and `Log to disk` in `Advanced` tab and replicate your bug, then attach the `AerialLog.txt` file (You can access this file through the Advanced tab).

#### Description of issue / Feature request

macOS Sonoma 14.0
2021 24", 4-Port iMac
Single (built-in) monitor.
Aerial 3.3.4
Aerial Companion 1.5.2.

Black screen still appears after indeterminate amount of time running Aerial screensaver.  The only thing that displays are MOST of the overlays (in my configuration (date, time, location) with the exception of:

Weather overlay cannot display based on location,  Aerial (legacy screensaver) does not appear to request location services from MacOS as it once did.  The only way I know to display it is to manually enter location (zip code or location name).

**Comment by glouel** [comment]
Please make sure you don't open multiple threads about the same issues. You already opened one about your black screen issue. 

When it comes to weather, for it to work you need this : 
- Aerial Companion 1.5.2 in your menu bar
- Aerial Companion allowed to use location services in System Settings (to trigger this, you can open Aerial settings from Companion, not from System Settings, go to overlays and press the test button, you should get prompted)
- Aerial 3.3.4

So it's likely Companion cannot pull your location and give it to the screensaver when asked.

**Comment by RGM1956** [comment]
Ariel Companion and Ariel are both set as advised. macOS Location Services is not being prompted to allow Ariel (or “legacy screensaver”) location information (as usual). This is a relatively new problem. It worked until recently. On Oct 18, 2023, at 6:02 AM, Guillaume Louel ***@***.***> wrote:﻿
Please make sure you don't open multiple threads about the same issues. You already opened one about your black screen issue.
When it comes to weather, for it to work you need this :

Aerial Companion 1.5.2 in your menu bar
Aerial Companion allowed to use location services in System Settings (to trigger this, you can open Aerial settings from Companion, not from System Settings, go to overlays and press the test button, you should get prompted)
Aerial 3.3.4

So it's likely Companion cannot pull your location and give it to the screensaver when asked.

**Comment by RGM1956** [comment]
Companion is set for current location, however, the screensaver is not and appears to not be requesting it.On Oct 18, 2023, at 6:17 AM, Ric Melton ***@***.***> wrote:﻿Ariel Companion and Ariel are both set as advised. macOS Location Services is not being prompted to allow Ariel (or “legacy screensaver”) location information (as usual). This is a relatively new problem. It worked until recently. On Oct 18, 2023, at 6:02 AM, Guillaume Louel ***@***.***> wrote:﻿
Please make sure you don't open multiple threads about the same issues. You already opened one about your black screen issue.
When it comes to weather, for it to work you need this :

Aerial Companion 1.5.2 in your menu bar
Aerial Companion allowed to use location services in System Settings (to trigger this, you can open Aerial settings from Companion, not from System Settings, go to overlays and press the test button, you should get prompted)
Aerial 3.3.4

So it's likely Companion cannot pull your location and give it to the screensaver when asked.

**Comment by RGM1956** [comment]
Pardon me.  You were right.  I did not have Ariel Companion resident in the met bar.  I don’t recall this being a prerequisite before.  Thank you.

**Comment by glouel** [comment]
No problem, Apple blocked screen savers to request location in Sonoma, and that's the core of the issue, the screen saver will never receive your location. 

Because of this, the workaround I've added is that when you launch the screensaver, it will request Companion to ask for your location in the background. Then it will send it back to the screensaver. 

It's a bit of a pain but it's the only workaround I can have for now. 

Regarding your black screen issue, I've closed the other one, let's try and fix it here. 

- First, can you tell me if the clock still runs or not when you see the black screen ? 
- Second, can you make a screenshot of your Lock Screen settings in System Settings (don't send it by email, you have to attach it in GitHub).

Then we'll go from here.

**Comment by RGM1956** [comment]
Those bastards really piss me off sometimes. 

Please forgive me for opening another ticket regarding the black screen issue.  I haven’t heard anything lately and assumed the ticket was closed.

**Comment by glouel** [comment]
No worries, can you provide the details I asked above ? Thanks

**Comment by RGM1956** [comment]
Ariel and Ariel Companion are both the versions you asked. The settings are all as you mentioned. I had to allow Ariel Companion to reside in the menu bar. It’s working now. I will have to wait until the next black screen in order to provide further details.  I believe the clock still works, but am not sure. On Oct 18, 2023, at 6:39 AM, Guillaume Louel ***@***.***> wrote:﻿
No worries, can you provide the details I asked above ? Thanks

**Comment by glouel** [comment]
This is what I need for your black screen issue :

- First, can you tell me if the clock still runs or not when you see the black screen ?
- Second, can you make a screenshot of your Lock Screen settings in System Settings (don't send it by email, you have to attach it in GitHub).

**Comment by peterchabs** [comment]
Hi @glouel , I sometimes get the black screen as well. Here are my specs along with the lock screen settings you were requesting from @RGM1956 
I believe the clock still appears through the black screen, the weather most definitely does.
Original release M1 MBA
<img width="485" alt="Screenshot 2023-11-23 at 11 41 32 PM" src="https://github.com/JohnCoates/Aerial/assets/26359726/e2b64c0b-e76b-48ec-a073-b0526d50c900">
<img width="712" alt="Screenshot 2023-11-23 at 11 42 58 PM" src="https://github.com/JohnCoates/Aerial/assets/26359726/d0441cc1-2530-4d79-94c9-00e516caafea">
<img width="279" alt="Screenshot 2023-11-23 at 11 43 15 PM" src="https://github.com/JohnCoates/Aerial/assets/26359726/91968d9a-ca8f-4f7e-b89f-4d5dd165b729">

**Comment by peterchabs** [comment]
![image](https://github.com/JohnCoates/Aerial/assets/26359726/4e3db29d-f883-4589-b682-a7ca5d408568)

**Comment by glouel** [comment]
Thanks @peterchabs that's helpful, and sorry about the issue.

Do you get them instantly, mid video, instead of a new video ? 

Also, if you leave it for 5 mins, does it stay black or a new one pops ? 

If you can, go to Aerial's settings, advanced, check "Debug Log". Then next time you get this, go back there, and grab the log (called AerialLog.txt) and send it to me with the rough time it happenned ? 

Let's see if we can sort it out !

**Comment by RGM1956** [comment]
I still see it from time to time as well. Frankly, with Sonoma‘s new  Wallpaper/Screensavers, I must question if this is an intentional“bomb” Apple laid in order to squash competition. We saw it when they stopped allowing Aerial Screensavers to access location information for the weather overlay. Sometimes, I hate Apple software. On Nov 24, 2023, at 6:31 AM, peterchabs ***@***.***> wrote:﻿

**Comment by peterchabs** [comment]
> Thanks @peterchabs that's helpful, and sorry about the issue.
> 
> Do you get them instantly, mid video, instead of a new video ?
> 
> Also, if you leave it for 5 mins, does it stay black or a new one pops ?
> 
> If you can, go to Aerial's settings, advanced, check "Debug Log". Then next time you get this, go back there, and grab the log (called AerialLog.txt) and send it to me with the rough time it happenned ?
> 
> Let's see if we can sort it out !

No need to apologize for anything, I appreciate the work you guys have put into this. Thank you.

This happens instead of a new video starting, and it does not go away unless I wake the mac from the screensaver.

I've attached a log file where it did happen during this time frame, scanning through it quickly I didn't notice anything alarming but i could be missing something!
[AerialLog.txt](https://github.com/JohnCoates/Aerial/files/13462934/AerialLog.txt)

**Comment by glouel** [comment]
@peterchabs looks like you added some custom videos in a forbidden folder. Can you make a screenshot of your "More videos", scrolled down, so it shows your local sources ? 

See, this path will never work because of security restrictions, when Aerial runs as a screensaver, it will never be allowed to load files from this path :
 
```
file:///Users/user/Documents/Aerial%20Videos/
```

I need to find a way to reject those paths when people add them but basically, you should only put videos in `/Users/Shared/` as this is one of the only "safe" place right now. Also, make sure that when you add the source,  you do it going from System Settings, not via Companion. Adding a local source via companion right now may cause issues too under Sonoma. 

So try removing your source, put your folder in `/Users/Shared/` and then let me know if it acts up again.

**Comment by peterchabs** [comment]
> @peterchabs looks like you added some custom videos in a forbidden folder. Can you make a screenshot of your "More videos", scrolled down, so it shows your local sources ? 
> 
> 
> 
> See, this path will never work because of security restrictions, when Aerial runs as a screensaver, it will never be allowed to load files from this path :
> 
>  
> 
> ```
> 
> file:///Users/user/Documents/Aerial%20Videos/
> 
> ```
> 
> 
> 
> I need to find a way to reject those paths when people add them but basically, you should only put videos in `/Users/Shared/` as this is one of the only "safe" place right now. Also, make sure that when you add the source,  you do it going into Aerial's settings from System Settings, not via Companion. Adding a local source via companion right now may cause issues too under Sonoma. 
> 
> 
> 
> So try removing your source, put your folder in `/Users/Shared/` and then let me know if it acts up again. It's highly likely it tried to load, was denied, and just got stuck because of it.

Ahh got it, thank you!

**Comment by RGM1956** [comment]
How can it be that after all these months this is still an issue?


￼

**Comment by amcclay** [comment]
Same issue here.  Weather overlay stopped working about 2 weeks ago. Debug log shows this:

```
2024-04-27 13:40:53.159 : fetching fresh weather
2024-04-27 13:40:53.159 : Starting Location initialization
2024-04-27 13:40:53.159 : Location using CompanionBridge data
2024-04-27 13:40:53.159 : === OW: Starting locationMode
2024-04-27 13:40:53.159 : Best matched language : en
2024-04-27 13:40:53.159 : 🎧 Adding music callback
2024-04-27 13:40:53.160 : 🎧🟧 manual triggering
2024-04-27 13:40:53.160 : 🎧 media remote callback
2024-04-27 13:40:53.161 : 🎧 audio info
2024-04-27 13:40:53.161 : 🎧  -  () without artwork
2024-04-27 13:40:53.162 : 🎧🟧 updateStatus
2024-04-27 13:40:53.204 : Starting 1 streams with 299146103 each, for content length of 299146103
2024-04-27 13:40:53.204 : last stream range: {0, 299146103}
2024-04-27 13:40:53.204 : Starting download for range bytes=0-299146103
2024-04-27 13:40:53.222 : The operation couldn’t be completed. (Aerial.NetworkError error 3.)

```
Is this an ongoing / known issue still?

**Comment by glouel** [comment]
> Same issue here. Weather overlay stopped working about 2 weeks ago. Debug log shows this:

Is it like **never** working since 2 weeks ago ? For some it sporadically doesn't work but it's not all the time. The weather API sometimes doesn't respond but it should be a rare occurence, only once in a while.

> ```
> 2024-04-27 13:40:53.159 : fetching fresh weather
> 2024-04-27 13:40:53.159 : Starting Location initialization
> 2024-04-27 13:40:53.159 : Location using CompanionBridge data
> 2024-04-27 13:40:53.159 : === OW: Starting locationMode

Ok so that part looks good, as far as I know. Can you confirm you have Aerial Companion running ? 

> ```
> 2024-04-27 13:40:53.204 : Starting 1 streams with 299146103 each, for content length of 299146103
> 2024-04-27 13:40:53.204 : last stream range: {0, 299146103}
> 2024-04-27 13:40:53.204 : Starting download for range bytes=0-299146103
> 2024-04-27 13:40:53.222 : The operation couldn’t be completed. (Aerial.NetworkError error 3.)

The network error is weird. Here it fails trying to get a video, but it's likely the weather failed too (maybe later in the log). 

Are you using some sort of firewall software ? 

If you can, try opening System Settings > ScreenSaver, make sure it says Aerial on top and press "Options" to open Aerial's settings. Then go to Overlays > Weather and try the "Test" button. It should open a window and show your found City. 
 
![Capture d’écran 2024-04-27 à 23 05 01](https://github.com/JohnCoates/Aerial/assets/37544189/812cc8ac-a953-4c04-8747-6efb9342efd9)

Let me know if it works here or not, and if it doesn't, can you paste that part of the log from that test too ? Thanks

**Comment by amcclay** [comment]
Sorry I should have also specified that both 'Use Current Location' and manually entering my city 'San Francisco, CA' result in 'City Not Found' (this worked a couple of weeks ago even with just my ZIP code)

<img width="793" alt="Screenshot 2024-04-27 at 2 15 42 PM" src="https://github.com/JohnCoates/Aerial/assets/15260379/0344f605-033e-4e7e-86f6-f1b6f195b65e">

Also, Aerial Companion is running and is approved in location services.

**Comment by glouel** [comment]
@amcclay Ok I know it's weird for the US, but don't use state abbrevations like CA, because the API doesn't understand it. 

Can you try with just "San Francisco" ? 

I have the same error with CA, and it works fine without.

**Comment by amcclay** [comment]
I feel dumb for not noticing the message that I should NOT be using state abbreviations, but I _did_ try it with and without them while troubleshooting and still get the City Not Found message.  

<img width="543" alt="Screenshot 2024-04-27 at 2 33 08 PM" src="https://github.com/JohnCoates/Aerial/assets/15260379/8036df68-7c54-44c9-a3b0-adf3d1594460">

Similarly, 'use location' returns the same message.

**Comment by glouel** [comment]
- Ok can you get me a log of those tries ? Maybe it shows this when there is a network error. It shouldn't but I had to change some stuff for Sonoma and it likely causes this.  

- Also which macOS version are you using exactly ?  

> Also note that I DO have Little Snitch installed, but it is completely disabled.

You may want to triple check Little Snitch as it may *still* block things. I don't use it but I know that some people thought they had disabled it but not completely disabled somehow. 

Because of the way it works now, Aerial doesn't do the network request directly, it goes though one of three processes : 

So make sure System Settings, Companion and ScreenSaverEngine (maybe legacyScreenSaver.appex) can all access network.

- You can try opening Settings from Companion and test there <- in that case it's Companion that should be allowed)
- Settings from System Settings <- System settings should be allowed (or maybe legacyScreenSaver)
- Screensaver itself <- ScreenSaverEngine.app (or maybe legacyScreenSaver)

Also, you can try grabbing a video manually from Aerial. If that fails too, it's definitely a network issue. Keep me posted.

**Comment by amcclay** [comment]
So, Little Snitch is as completely disabled as it can be without me totally uninstalling it (which I may try down the line)

Aerial Companion and legacyScreenSaver are approved in Location but there is no  entry there for ScreenSaverEngine.app.  Are you referring solely to Location access or is there another setting I should look at?

Manually downloading view the 'More Videos' seems to work fine (it downloaded the Joshua Michaels videos without issue)

**Comment by glouel** [comment]
> So, Little Snitch is as completely disabled as it can be without me totally uninstalling it (which I may try down the line)

Ok, as I understand there's a separate settings for Apps and System. But see below.
 
> Aerial Companion and legacyScreenSaver are approved in Location but there is no entry there for ScreenSaverEngine.app. Are you referring solely to Location access or is there another setting I should look at?

I'm referring specifically to Networking issues. 

> Manually downloading view the 'More Videos' seems to work fine (it downloaded the Joshua Michaels videos without issue)

Was that via opening Aerial's Settings via Companion or System Settings ? And via the same way, Weather still fails when you press Test button ? 

Companion should be the easiest one to try out. 

So in case that's not what you did, please try both downloading and the Test button on weather (with San Francisco) and see what it does. And if only one fails, again please please attach a log of that whole testing (both the one that worked, downloading video, and the one that fails, weather) ? 

I'll try to have a look tomorrow.  

Also, are you using any form of proxy or vpn ?

**Comment by amcclay** [comment]
I'll run all these tests later tonight.

I should say, that this failure to access the weather happened on TWO Macs I have in the same apartment, which leads me to believe it is likely a network issue somewhere.  Possibly even some kind of fireballing happening at the level of my ISP. 

One thing I may try is bringing my laptop to a cafe and seeing if I am able to access the weather data there.

Thanks for all your help so far!

**Comment by glouel** [comment]
No problem, and I agree it seems like a definite network issue, if it's not on your mac, could be a router, or ISP blocking. Or maybe somehow your IP range is blocked from the API, but that sounds super weird.  

If you have some vpn provider that lets you connect elsewhere, it's also a good quick way to bypass all that. 

Couple more thoughts : 
- Check if https://openweathermap.org works for you (it's their front facing website)
- Openweather's API uses HTTP, and not HTTP**S**. Make sure you don't have anything in front of your mac (router, some other filtering) that could deny those ? ISP seems unlikely but who knows ? 
- http://api.openweathermap.org/data/2.5/weather is one of the endpoint we use. It does require an API key so you won't be able to fully test it, but give it a shot in Safari for example, see if that works. 

Opening that link in Safari should return a JSON like this :

```
{"cod":401, "message": "Invalid API key. Please see https://openweathermap.org/faq#error401 for more info."}
``` 

Best of luck and keep me posted, I'll try to get back to you tomorrow, take care.

**Comment by RGM1956** [comment]
I hate to burst bubbles, but the weather overlay STOPPED working on my system too.  And I was not using state abbreviations.  

As of very recently it stopped displaying current weather with or without the state being fully spelled out. And cannot find the city regardless of advice given.

 I give up.  Ariel screensavers have been one problem after another spice the release of Sonoma, several months ago, and isn’t  worth it.  I’m uninstalling it.

**Comment by glouel** [comment]
@amcclay let me know if 

http://api.openweathermap.org/data/2.5/weather

and 

https://api.openweathermap.org/data/2.5/weather

returns the json above in a browser. OpenWeather used to not support https but it looks like they added it recently. I'll transition to https in next build but it could be the issue.

**Comment by glouel** [comment]
@amcclay I release 3.3.8 with https support for weather, as this is now supported on their end. Maybe it will fix your issue, maybe not, in any case keep me posted. If you test via Companion, you need to restart companion after you update. 

This also includes some fixes for failed video downloads which may cause black screens with overlays.

**Comment by amcclay** [comment]
Lol just came here to say weather re-appeared on both of my Macs and saw your response.  You did indeed fix it on my end.

**THANK YOU!**

I really do rely on the weather as I have Aerial "always on" on a small screen in my living room which lets me know what jacket to wear before leaving my apartment. 

Again, many thanks for being such a responsive and proactive developer.

**Comment by glouel** [comment]
@amcclay awesome, thanks for reporting it. I'm still not sure where the issue was, maybe they are starting to phase out their http only servers and depending on CDNs, it always failed. Or something somewhere blocked it. Anyway, take care and enjoy !

**Comment by amcclay** [comment]
Oh before I forget (I'm sure you've been asked this before) what is the possibility of Aerial being released for Apple TV?  I realize that Apple wouldn't allow something that pulls their own screensavers down, but the non-apple videos in Aerial are honestly just as good if not better now.  

If it were released for Apple TV, I'd easily pay $5 for it, and I'd guess others would to as it would replace many other apps (weather, clock, aerial, etc) - I could also see it being used as a kiosk display since so many bars / businesses use Apple TV.

Anyway, just a thought.

**Comment by glouel** [comment]
> Oh before I forget (I'm sure you've been asked this before) what is the possibility of Aerial being released for Apple TV? I realize that Apple wouldn't allow something that pulls their own screensavers down, but the non-apple videos in Aerial are honestly just as good if not better now.
> 
> If it were released for Apple TV, I'd easily pay $5 for it, and I'd guess others would to as it would replace many other apps (weather, clock, aerial, etc) - I could also see it being used as a kiosk display since so many bars / businesses use Apple TV.
> 
> Anyway, just a thought.

So you can't replace the Apple TV screensaver, only the built in one works. 

As a standalone app, it's probably possible but would likely require many changes. The Apple TV doesn't have a lot of storage so that's the main issue, you can't cache many videos, it's severely restricted, and your cache may  get wiped at any moment. 

If I have some time, maybe I'll have another look someday, but the restrictions seemed so harsh that I'm not sure it's possible. And that's setting aside if Apple would even allow it on the store which is hard to know.

**Comment by amcclay** [comment]
Ah yes. The video cache! I hadn’t even thought of that lol. 

and yes I also realize Apple (stupidly) doesn’t allow you to replace their screensavers. 

Still, even just as a “slideshow” app with the rest of the functionality of overlays I could see it being very popular as a kiosk display. 

Like a business could make a logo and still have info like time and weather to draw attention to it. 

But as I think about it, just dealing with the UI to actually customize the overlays and other settings would be a huge amount of work so … nevermind haha
